api/types.proto

```diff
@@ -281,15 +281,32 @@ message UpdateConfig {
 	enum FailureAction {
 		PAUSE = 0;
 		CONTINUE = 1;
-		// TODO(aaronl): Add ROLLBACK as a supported failure mode.
-		// (#486)
+		ROLLBACK = 2;
 	}
 
 	// FailureAction is the action to take when an update failures.
-	// Currently, a failure is defined as a single updated task failing to
-	// reach the RUNNING state. In the future, there will be configuration
-	// to define what is treated as a failure (see #486 for a proposal).
 	FailureAction failure_action = 3;
+
+	// AllowedFailureFraction is the fraction of tasks that may fail during
+	// an update before the failure action is invoked. Any task created by
+	// the current update which ends up in one of the states REJECTED,
+	// COMPLETED or FAILED while the update is still running is counted as
+	// a failure. The number of failures is divided by the number of tasks
```

Comment:
> How do you know which failures belong to which update?

My plan is to do it by comparing task specs. If a task failed, and its spec matched the one we're updating to, then we would conclude the task was started by the current update.

I think this is the right approach because it's very simple and doesn't require keeping any local state that would be lost during a manager failover.